pennylane/interfaces/jacobian_products.py

```diff
@@ -0,0 +1,176 @@
+        True
+
+        """
+
+    @abc.abstractmethod
+    def compute_jacobian(self, tapes: Batch) -> Tuple:
+        """Compute the full jacobian for a batch of tapes.
+
+        This method is required to compute Jacobians in the ``jax-jit`` interface
+
+        Args:
+            tapes: the batch of tapes to take the jacobian of
+
+        >>> tape0 = qml.tape.QuantumScript([qml.RX(0.1, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> tape1 = qml.tape.QuantumScript([qml.RY(0.2, wires=0)], [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))])
+        >>> batch = (tape0, tape1)
+        >>> jp_method.compute_jacobian(batch)
+        (array(-0.09983342), (array(-0.19866933), array(0.98006658)))
+
+        """
+
+
+class TransformJacobianProducts(JacobianProductCalculator):
+    """Compute VJPs, JVPs and Jacobians via a :class:`~.gradient_transform`.
+
+    Args:
+        inner_execute (Callable[[Tuple[QuantumTape]], ResultBatch]): a function that
+            executes the batch of circuits and returns their results.
+        gradient_transform (pennylane.gradients.gradient_transform): the gradient transform to use.
+        gradient_kwargs (dict): Any keyword arguments for the gradient transform.
+
+    >>> inner_execute = qml.device('default.qubit').execute
+    >>> gradient_transform = qml.gradients.param_shift
+    >>> kwargs = {"broadcast": True}
+    >>> jp_method = TransformJacobianProducts(inner_execute, gradient_transform, kwargs)
+
+    """
+
+    def __repr__(self):
+        return f"TransformJacobianProducts({self._inner_execute}, gradient_transform={self._gradient_transform}, gradient_kwargs={self._gradient_kwargs})"
+
+    def __init__(
+        self,
+        inner_execute: Callable,
+        gradient_transform: "qml.gradients.gradient_transform",
+        gradient_kwargs: Optional[dict] = None,
+    ):
+        self._inner_execute = inner_execute
+        self._gradient_transform = gradient_transform
+        self._gradient_kwargs = gradient_kwargs or {}
+
+    def execute_and_compute_jvp(self, tapes: Batch, tangents: Tuple[Tuple[TensorLike]]):
+        num_result_tapes = len(tapes)
+
+        jvp_tapes, jvp_processing_fn = qml.gradients.batch_jvp(
+            tapes, tangents, self._gradient_transform, gradient_kwargs=self._gradient_kwargs
+        )
+
+        full_batch = tapes + tuple(jvp_tapes)
+
+        full_results = self._inner_execute(full_batch)
+
+        results = full_results[:num_result_tapes]
+        jvp_results = full_results[num_result_tapes:]
+        jvps = jvp_processing_fn(jvp_results)
+        return tuple(results), tuple(jvps)
+
+    def compute_vjp(self, tapes: Batch, dy: Tuple[Tuple[TensorLike]]):
+        vjp_tapes, processing_fn = qml.gradients.batch_vjp(
+            tapes, dy, self._gradient_transform, gradient_kwargs=self._gradient_kwargs
+        )
+
+        vjp_results = self._inner_execute(vjp_tapes)
+        return tuple(processing_fn(vjp_results))
```

Comment:
In contrast to `batch_jvp.processing_fn`, `batch_vjp.processing_fn` indeed does not cast to `tuple`.
Just mentioning it, not sure whether we want to assimilate this? :thinking: